File: app/src/main/java/algot/emil/persistence/Weather.java
package algot.emil.persistence;


import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Weather {

    @PrimaryKey
    public int uid;

    @ColumnInfo(name = "time")
    public String time;

    @ColumnInfo(name = "temperature_2m_max")
    public float temperature;


    @ColumnInfo(name = "rain_sum")
    public String rainSum;

    @ColumnInfo(name = "showers_sum")
    public String showersSum;

    @ColumnInfo(name = "snowfall_sum")
    public String snowfallSum;




}

File: app/src/main/java/algot/emil/persistence/Weather.kt
package algot.emil.persistence

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey


@Entity
data class Weather(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    @ColumnInfo(name = "time") val time: String,
    @ColumnInfo(name = "temperature_2m_max") val temperature: Float,
    @ColumnInfo(name = "rain_sum") val rainSum: Float,
    @ColumnInfo(name = "showers_sum") val showersSum: Float,
    @ColumnInfo(name = "snowfall_sum") val snowfallSum: Float
)